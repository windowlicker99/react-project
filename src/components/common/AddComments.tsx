import React, { ChangeEventHandler, FC, useState } from 'react';
import { getIn, useFormikContext } from 'formik';
import { Textarea } from '@components/common/form-controls/Textarea';
import { Btn } from '@components/common/form-controls/Btn';

interface AddCommentProps {
  name: string;
}

export const AddComments: FC<AddCommentProps> = ({ name }) => {
  const { errors, touched, setFieldValue } = useFormikContext();
  const [comment, setComment] = useState('');
  const handleChangeFieldValue: ChangeEventHandler<HTMLTextAreaElement> = (e) => {
    const {
      target: { value },
    } = e;
    setComment(value);
  };
  const handleResetComment = () => setComment('');
  const handleApplyComment = () => setFieldValue(name, comment);
  return (
    <>
      <Textarea
        placeholder="Add a message"
        error={getIn(touched, name) && getIn(errors, name)}
        value={comment}
        name={name}
        onChange={handleChangeFieldValue}
      />
      <div className="btn-group mt-30 flex-end">
        <Btn btnStatus="btn-lg-outline mr-15" btnText="Cancel" onClick={handleResetComment} />
        <Btn btnStatus="btn-lg-primary" btnText="Apply" disabledBtn={!comment} onClick={handleApplyComment} />
      </div>
    </>
  );
};
