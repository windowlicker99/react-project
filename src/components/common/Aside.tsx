import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IAsideProps {
  header?: ReactNode;
  asideStatus?: string;
  footer?: ReactNode;
}

export const Aside: FC<IAsideProps> = ({ asideStatus = '', header, footer, children }) => (
  <div className={classNames('aside', asideStatus)}>
    <div className="aside-header">{header}</div>
    <div className="aside-body">{children}</div>
    <div className="aside-footer"> {footer}</div>
  </div>
);
