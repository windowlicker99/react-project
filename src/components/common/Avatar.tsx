import React, { FC } from 'react';
import AvatarImg from '@images/avatar.jpeg';

interface IAvatarProps {
  name: string;
  avatar: string;
}

export const Avatar: FC<IAvatarProps> = ({ name, avatar }) => (
  <div className="avatar-card">
    <div className="avatar">
      <img className="avatar-img" src={avatar || AvatarImg} alt="avatar" />
    </div>
    <span className="avatar-name">{name || 'John Stone'}</span>
  </div>
);
