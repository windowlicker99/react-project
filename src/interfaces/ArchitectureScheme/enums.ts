export enum EElementPropName {
  node = 'node',
  connector = 'connector',
}
