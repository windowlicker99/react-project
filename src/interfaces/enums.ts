export enum ERiskLevels {
  veryLow = 'Very Low',
  low = 'Low',
  medium = 'Medium',
  moderate = 'Moderate',
  high = 'High',
  veryHigh = 'Very High',
  noData = 'No Data',
}

export enum EBadgeTypes {
  veryLow = 'success',
  low = 'primary',
  medium = 'warning',
  moderate = 'warning',
  high = 'warning-danger',
  veryHigh = 'danger',
  noData = 'default',
}

export enum ETicketsTypes {
  completment = 'completment',
  riskAssessment = 'riskAssessment',
  implementation = 'implementation',
}

export enum EAside {
  gateway = 'gateways',
  domain = 'domains',
  ecu = 'ecus',
  ECUasGateway = 'ECUasGateway',
  subEcu = 'subEcu',
}

export enum EAsideActualElement {
  gateways = 'gateways',
  domains = 'domains',
  ecus = 'ecus',
  ECUasGateway = 'ecus',
  subEcu = 'ecus',
}

export enum EConnectivity {
  Ethernet = 'Ethernet',
  CAN = 'CAN',
  LIN = 'LIN',
}

export enum EUpdateProcessRadio {
  managing = 'managing',
  using = 'using',
}

export enum EDirectCommunication {
  WiFi = 'Wi-Fi',
  bluetooth = 'Bluetooth',
  NFC = 'NFC',
  BLE = 'BLE',
  GPS = 'GPS/GNSS',
  cellularConnection = 'Cellular Connection',
  V2X = 'V2X',
}

export enum EElementCreationTabs {
  create = 'Create New',
  choose = 'Choose from Collection',
}

export enum EInitiatePrivacyLawImpact {
  Severe = 'Severe | The privacy damage leads to significant or even irreversible impact to the road user',
  Major = 'Major | The privacy damage leads to serious impact to the road user',
  Moderate = 'Moderate | The privacy damage leads to significant inconveniences to the road user ',
  Negligible = 'Negligible | The privacy damage leads to no effect or can create few inconveniencies to the road user',
  None = 'No privacy / law impact',
}

export enum EInitiateOperationalImpact {
  Severe = 'Severe | The operational damage leads to a vehicle not working, from non-intended operation up to the vehicle being non-operational',
  Major = 'Major | The operational damage leads to the loss of a vehicle function',
  Moderate = 'Moderate | The operational damage leads to partial degradation of a vehicle function or performance',
  Negligible = 'Negligible | The operational damage leads to no effect or indiscernible degradation of a vehicle function or performance',
  None = 'No operational impact',
}

export enum EInitiateFinancialImpact {
  Severe = 'Severe | The financial damage leads to catastrophic consequences which the affected stakeholder might not overcome',
  Major = 'Major | The financial damage leads to substantial consequences which the affected stakeholder will be able to overcome',
  Moderate = 'Moderate | The financial damage leads to inconvenient consequences which the affected stakeholder will be able to overcome with limited resources',
  Negligible = 'Negligible | The financial damage leads to no effect, negligible consequences or is irrelevant to the stakeholder',
  None = 'No financial impact',
}

export enum EInitiateSafetyImpact {
  Severe = 'Severe | Life-threatening injuries (survival uncertain), fatal injuries ',
  Major = 'Major | Severe and life-threatening injuries (survival probable)',
  Moderate = 'Moderate | Light and moderate injuries',
  Negligible = 'Negligible | No injuries',
  None = 'No safety impact',
}

export enum EController {
  id = 'id',
  name = 'name',
  safetyCriticalityLevel = 'safetyCriticalityLevel',
  firmwareVersion = 'firmwareVersion',
  onlineConnectivity = 'onlineConnectivity',
  vehicleId = 'vehicleId',
  operationSystem = 'operationSystem',
}

export enum EDomain {
  id = 'id',
  model = 'model',
  platform = 'platform',
  name = 'name',
  domainUsage = 'domainUsage',
  communication = 'communication',
  onlineConnectivity = 'onlineConnectivity',
  color = 'color',
}

export enum EECU {
  id = 'id',
  name = 'name',
  type = 'type',
  complianceLevel = 'complianceLevel',
  riskLevel = 'riskLevel',
  operationSystem = 'operationSystem',
  firmwareVersion = 'firmwareVersion',
  onlineConnectivity = 'onlineConnectivity',
  assignedTo = 'assignedTo',
  supplier = 'supplier',
  safetyCriticalityLevel = 'safetyCriticalityLevel',
  domain = 'domains',
  connectedECUs = 'connectedECUs',
}

export enum EApproved {
  approved = 'Approved',
  notApproved = 'Not Approved',
}

export enum EImplemented {
  implemented = 'Implemented',
  notImplemented = 'Not Implemented',
}

export enum EPhaseStatus {
  active = 'active',
  inProgress = 'in-progress',
}

export enum EPhase {
  development = 'Development',
  testing = 'Testing',
  compliancy = 'Compliancy',
  production = 'Production',
}

export enum ERegulations {
  wp29 = 'UNECE WP.29',
}

export enum ESafetyCriticalityLevel {
  none = 'None',
  asilA = 'ASIL A',
  asilB = 'ASIL B',
  asilC = 'ASIL C',
  asilD = 'ASIL D',
}

export enum EECUTypes {
  gateway = 'Gateway',
  ethernetSwitch = 'Ethernet Switch',
  hostECU = 'Host',
  VM = 'VM',
  regular = 'Regular',
}

export enum EProgressBarTypes {
  vertical = 'vertical',
  horizontal = 'horizontal',
}

export enum ESecurityTypes {
  concept = 'concept',
  requirements = 'requirements',
  penetrationTests = 'penetrationTests',
}
export enum EThreatsControlsPath {
  threats = 'threats',
  controls = 'controls',
}

export enum EFindings {
  main = 'Main Findings',
  detailed = 'Detailed Findings',
}

export enum EThreatTabs {
  UNECE = 'UNECE',
  OEM = 'OEM',
}

export enum ETicketStatus {
  open = 'open',
  progress = 'progress',
  closed = 'closed',
  deleted = 'deleted',
}

export enum EExpertise {
  layman = 'Layman',
  proficient = 'Proficient',
  expert = 'Expert',
  multipleExperts = 'Multiple Experts',
}

export enum EAccess {
  unlimited = 'Unlimited',
  easy = 'Easy',
  moderate = 'Moderate',
  difficult = 'Difficult',
  multipleExperts = 'Multiple Experts',
}

export enum EEquipment {
  standard = 'Standard',
  specialized = 'Specialized',
  bespoke = 'Bespoke',
  multipleBespoke = 'Multiple bespoke',
}

export enum EAttackDifficultyFinalScope {
  veryHigh = 'Very High / Basic Attack',
  high = 'High / Enhanced-Basic Attack',
  medium = 'Medium / Moderate Attack',
  low = 'Low / Difficult Attack',
  veryLow = 'Very Low / Beyond Difficult Attack',
}

export enum EComponentStatuses {
  large = 'large',
  usual = 'usual',
}

export enum EBadgeText {
  primary = 'Low',
  danger = 'Very High',
  success = 'Very Low',
  warning = 'Moderate',
}

export enum EIRASteps {
  step1 = 1,
  step2 = 2,
  step3 = 3,
  step4 = 4,
  step5 = 5,
  step6 = 6,
}

export enum EGatewayTypes {
  gateway = 'gateway',
  ecu = 'ECUasGateway',
}
