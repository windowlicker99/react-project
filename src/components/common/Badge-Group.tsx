import React, { FC } from 'react';
import { EBadgeText } from '@interfaces/enums';
import { Badge } from './Badge';

interface IProps {
  type: string;
}
export const BadgeGroup: FC<IProps> = ({ type }) => (
  <div className="badge-group">
    <span className="badge-group-labe">Risk Level</span>
    <Badge badgeType={type} badgeText={EBadgeText[type as keyof typeof EBadgeText]} />
  </div>
);
