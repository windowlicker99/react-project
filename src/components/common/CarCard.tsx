import React, { FC } from 'react';
import { EPhase } from '@interfaces/enums';

interface ICarCardProps {
  name: string;
  logo: string;
  photo: string;
  model: string;
  platform: string;
  currentPhase: EPhase;
}

export const CarCard: FC<ICarCardProps> = ({ logo, photo, model, name, currentPhase, platform }) => (
  <div className="car-card">
    <div className="car-info-img">
      <img alt="car-img" className="car-img" src={photo} />
      <img className="car-logo" alt="" src={logo} />
    </div>
    <div className="car-info">
      <span className="car-name">{name}</span>
      <div className="car-info-item">
        <span className="car-property">Model: </span>
        <span className="car-value">{model}</span>
      </div>
      <div className="car-info-item">
        <span className="car-property">Platform: </span>
        <span className="car-value">{platform}</span>
      </div>
      <div className="car-info-item">
        <span className="car-property">Ð¡urrent Phase: </span>
        <span className="car-value">{currentPhase}</span>
      </div>
    </div>
  </div>
);
