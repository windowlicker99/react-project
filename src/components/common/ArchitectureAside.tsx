import React, { FC } from 'react';
import { Heading } from '@components/common/Heading';
import { Aside } from './Aside';
import { ICurrentElement } from '@/interfaces/interfaces';

interface IArchitectureAsideProps {
  asideStatus?: string;
  currentElement: ICurrentElement;
}

export const ArchitectureAside: FC<IArchitectureAsideProps> = ({ asideStatus = '', children, currentElement }) => (
  <Aside
    asideStatus={asideStatus}
    header={
      currentElement ? (
        <Heading headingText="Details" linkText="Edit details" linkIcon="icon-ellipsis-vertival" />
      ) : null
    }
  >
    {children}
  </Aside>
);
