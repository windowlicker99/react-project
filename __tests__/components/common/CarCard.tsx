import React from 'react';
import { render } from '@testing-library/react';
import { CarCard } from '@components/common/CarCard';
import { EPhase } from '@interfaces/enums';

describe('CardCard tests', () => {
  it('Should render', () => {
    const { getByText } = render(
      <CarCard platform="Platform" name="Audi" currentPhase={EPhase.development} logo="" photo="" model="test" />
    );
    const span = getByText('Audi');
    expect(span).toBeTruthy();
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <CarCard platform="Platform" name="test" currentPhase={EPhase.development} logo="" photo="" model="test" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
