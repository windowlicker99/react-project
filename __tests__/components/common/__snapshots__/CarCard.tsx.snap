// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCard tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="car-card"
  >
    <div
      class="car-info-img"
    >
      <img
        alt="car-img"
        class="car-img"
        src=""
      />
      <img
        alt=""
        class="car-logo"
        src=""
      />
    </div>
    <div
      class="car-info"
    >
      <span
        class="car-name"
      >
        test
      </span>
      <div
        class="car-info-item"
      >
        <span
          class="car-property"
        >
          Model: 
        </span>
        <span
          class="car-value"
        >
          test
        </span>
      </div>
      <div
        class="car-info-item"
      >
        <span
          class="car-property"
        >
          Platform: 
        </span>
        <span
          class="car-value"
        >
          Platform
        </span>
      </div>
      <div
        class="car-info-item"
      >
        <span
          class="car-property"
        >
          Ð¡urrent Phase: 
        </span>
        <span
          class="car-value"
        >
          Development
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
